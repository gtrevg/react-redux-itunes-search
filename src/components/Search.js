import React from 'react';
import { Link } from "react-router-dom";

const Search = () => {
  return <div><Link to="/detail">Go to detail</Link></div>
};

export default Search;
