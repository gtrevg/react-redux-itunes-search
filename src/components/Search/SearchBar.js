import React from 'react';
import { Input } from 'antd';

import './SearchBar.css';

const Search = Input.Search;

const SearchBar = () => {
  return (
    <div className="SearchBar">
      <Search
          placeholder="Search for artists, songs, albums, genre,... Whatever!"
          enterButton="Search"
          size="large"
          onSearch={value => console.log(value)} />
        </div>
  );
};

export default SearchBar;
