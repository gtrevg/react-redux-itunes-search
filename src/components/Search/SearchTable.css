.SearchTable {
  padding: 5px;
  margin: 10px;
  border-radius: 8px;
}

.SearchTableTitle {
    width: 25%;
    min-width: 160px;
}

.SearchTableArtist {
    width: 20%;
    min-width: 120px;
}

.SearchTableAlbum {
    width: 25%;
    min-width: 160px;
}

.SearchTableRelease {
    width: 5%;
    min-width: 120px;
}

.SearchTableCover {
    width: 5%;
    min-width: 80px;
}

.SearchTableLenght {
    width: 5%;
    min-width: 80px;
}

.SearchTableGenre {
    width: 10%;
    min-width: 120px;
}

.SearchTablePrice {
    width: 5%;
    min-width: 120px;
}

@media (max-width: 1240px) {
  .SearchTablePrice {
    display: none;
  }
}

@media (max-width: 1120px) {
  .SearchTableGenre {
    display: none;
  }
}

@media (max-width: 1040px) {
  .SearchTableLenght {
    display: none;
  }
}

@media (max-width: 960px) {
  .SearchTableCover {
    display: none;
  }
}

@media (max-width: 840px) {
  .SearchTableRelease {
    display: none;
  }
}
