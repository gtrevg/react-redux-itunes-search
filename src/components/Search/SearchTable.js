import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { Table } from 'antd';

import './SearchTable.css';

const columns = [
  {
    title: 'Title',
    dataIndex: 'title',
    key: 'title'
  },
  {
    title: 'Artist',
    dataIndex: 'artist',
    key: 'artist'
  },
  {
    title: 'Album title',
    dataIndex: 'album',
    key: 'album'
  },
  {
    title: 'Release date',
    dataIndex: 'release',
    key: 'release'
  },
  {
    title: 'Cover',
    dataIndex: 'cover',
    key: 'cover',
    render: (text, record) => (
      <img alt={record.title} src={text} style={{width: 60}}/>
    ),
  },
  {
    title: 'Lenght',
    dataIndex: 'lenght',
    key: 'lenght'
  },
  {
    title: 'Genre',
    dataIndex: 'genre',
    key: 'genre'
  },
  {
    title: 'Price',
    dataIndex: 'price',
    key: 'price'
  }
];

class SearchTable extends PureComponent {
  render() {
    const { isLoading, dataSource } = this.props;
    return (
      <div className="SearchTable">
        <Table
          loading={isLoading}
          dataSource={dataSource}
          columns={columns}
          pagination={{ pageSize: 5 }}
          onRow={(record) => {
            return {
              onClick: () => { console.log(record); },       // click row
              };
            }}
        />
      </div>
    );
  }
};

const mapStateToProps = (state) => {
  return {
    ...state.search
  }
}

export default connect(mapStateToProps)(SearchTable);
