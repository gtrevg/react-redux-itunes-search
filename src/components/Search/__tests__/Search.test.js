import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { mockComponent } from '../../../utils/testUtils';

import Search from '../index';

jest.mock('../SearchBar', () => {
	return props => mockComponent('SearchBar', props);
});

jest.mock('../SearchTable', () => {
	return props => mockComponent('SearchTable', props);
});

describe('Search component', () => {
	const component = <Search />;

	it('Renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(component, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('Snapshot matchs', () => {
		const tree = renderer.create(component).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
