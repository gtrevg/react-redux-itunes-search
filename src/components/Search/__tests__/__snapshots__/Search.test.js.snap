// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component Snapshot matchs 1`] = `
<div>
  <mocked
    name="SearchBar"
    props={Object {}}
  />
  <mocked
    name="SearchTable"
    props={Object {}}
  />
</div>
`;
